### PR TITLE
Try fetch font from cache first

### PR DESCRIPTION
**Pull request by bigfoodK**
Cache fonts explicitly with `namui::cache`.

### CODE REVIEW

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs [hunk 1]
```diff
@@ -26,32 +26,52 @@ async fn load_fallback_font_typefaces() -> Result<(), Box<dyn std::error::Error>
...
+                .map(|typeface| (*typeface_type, Arc::new(typeface)))
+        },
+    ))
+    .await?
+    .into_iter();
+    for (typeface_type, typeface) in iter {
         crate::typeface::load_typeface(&typeface_type, typeface.clone());
         load_default_font_of_typeface(typeface);
-    });
+    }
```
Comment (namse) [inline namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:47]: Could you make this code more simpler?

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs [hunk 2]
```diff
@@ -26,32 +26,52 @@ async fn load_fallback_font_typefaces() -> Result<(), Box<dyn std::error::Error>
...
     let url = crate::Url::parse("bundle:__system__/font/map.json")?;
-    let typeface_file_urls_file = crate::file::bundle::read_json(url).await?;
-
-    Ok(typeface_file_urls_file)
+    let Some(cached_typeface_file_urls_file) = crate::cache::get_serde(url.as_str()).await? else {
+        let typeface_file_urls_file = crate::file::bundle::read_json(url.clone()).await?;
+        crate::cache::set_serde(url.as_str(), &typeface_file_urls_file).await?;
+        return Ok(typeface_file_urls_file)
+    };
+    Ok(cached_typeface_file_urls_file)
```
Comment (namse) [inline namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:74]: If you want to cache this too, I suggest to make a function which load data from cache first and fetch if fail.

**Review by namse — CHANGES_REQUESTED** [review]

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs (hunk 1 above)
Comment (bigfoodK) [reply namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:47]: I think I can separate the `let iter`.

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs (hunk 2 above)
Comment (bigfoodK) [reply namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:74]: Let me check.

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs (hunk 1 above)
Comment (namse) [reply namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:47]: What do you mean?

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs [hunk 3]
```diff
@@ -26,32 +26,52 @@ async fn load_fallback_font_typefaces() -> Result<(), Box<dyn std::error::Error>
 
 async fn get_noto_color_emoji_typeface() -> Result<Typeface, Box<dyn std::error::Error>> {
     let url = crate::Url::parse("bundle:__system__/font/NotoColorEmoji.woff2")?;
-    let bytes = crate::file::bundle::read(&url)
-        .await
-        .expect(format!("Could not fetch {}", url).as_str());
-
-    Ok(Typeface::new(&bytes))
+    get_typeface(url).await
```
Comment (namse) [inline namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:29]: just, rollback this changes and make a change to the line of data fetch.

```rust
// before
let bytes = crate::file::bundle::read(&url).await

// after
let bytes = get_data_from_cache_first_then_bundle_blah_blah_some_good_name(&url).await
```

Then everything is same. You don't have to changes the structure of codes.

**Review by namse — CHANGES_REQUESTED** [review]

File: namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs (hunk 1 above)
Comment (namse) [reply namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:47]: What do you mean?

Comment (bigfoodK) [reply namui/src/namui/system/typeface/load_sans_typeface_of_all_languages.rs:47]: Rollbacking to previous structure.

**Review by namse — APPROVED** [review]
Cool. Simple is besto

**Pull request merged**